### PR TITLE
feat(build): package shared-openssl mongosh binaries MONGOSH-1224

### PR DESCRIPTION
**Pull request by addaleax**
Adjust the build configuration, and the package build steps,
to account for different package names being used in package
creation.

Extend the packaging and package test matrices to cover
shared-openssl artifacts. This does *not* include publishing
them as part of the release steps.

This also addresses MONGOSH-1213 (Linux arm64 smoke tests
are done inside docker now, just like x64) since it proves
convenient for testing rpm/deb packages that come with
extra dependencies (such as specific OpenSSL versions).

Additionally, as a drive-by fix, add Linux x64 and arm64
package smoke testing for the regular (static-OpenSSL)
.tgz tarballs.

### CODE REVIEW

File: config/build.conf.js [hunk 1]
```diff
@@ -151,10 +161,20 @@ module.exports = {
     },
     metadata: {
```
Comment (addaleax) [inline config/build.conf.js:162]: This may need to be restructured a bit for actually releasing these artifacts, but for now this works.

File: scripts/docker/rocky8-epel-rpm.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+FROM rockylinux:8
+
+ARG artifact_url=""
+ADD ${artifact_url} /tmp
+ADD node_modules /usr/share/mongodb-csfle-library-version/node_modules
+RUN dnf update -y
+# epel-release so that openssl3 is installable
```
Comment (addaleax) [inline scripts/docker/rocky8-epel-rpm.Dockerfile:7]: I considered just adding this to the regular Rocky Linux 8 Dockerfile, but decided against it since it’s nice to actually know that no special packages are required for install the static-openssl mongosh on the base system.

File: scripts/docker/rocky8-rpm.Dockerfile [hunk 3]
```diff
@@ -3,9 +3,9 @@ FROM rockylinux:8
 ARG artifact_url=""
 ADD ${artifact_url} /tmp
 ADD node_modules /usr/share/mongodb-csfle-library-version/node_modules
-RUN yum repolist
-RUN yum install -y man
-RUN yum install -y /tmp/*mongosh-*.x86_64.rpm
+RUN dnf repolist
+RUN dnf install -y man
+RUN dnf install -y /tmp/*mongosh*.rpm
```
Comment (addaleax) [inline scripts/docker/rocky8-rpm.Dockerfile:8]: Just to align with the other Dockerfiles for distros that have `dnf`

File: .evergreen/evergreen.yml.in [hunk 4]
```diff
@@ -1137,22 +1165,26 @@ buildvariants:
...
-      - name: pkg_test_docker_suse12_rpm
-      - name: pkg_test_docker_suse15_rpm
-      - name: pkg_test_docker_amazonlinux1_rpm
-      - name: pkg_test_docker_amazonlinux2_rpm
-      - name: pkg_test_docker_debian9_deb
-      - name: pkg_test_docker_debian10_deb
-      - name: pkg_test_docker_debian11_deb
-      - name: pkg_test_docker_ubuntu18_04_deb
-      - name: pkg_test_docker_ubuntu20_04_deb
+  <% for (const { distributionBuildVariants, executableOsId } of EXECUTABLE_PKG_INFO) {
```
Comment (mcasimir) [inline .evergreen/evergreen.yml.in:1172]: I was wondering, is there a way to move this heavy logic outside of yaml? Seems hard to understand the code here and also a bit intimidating to change.

If we are using something like lodash.template or similar I'd prefer something like this with more helpers:

```
<% for ( const dockerfile of getSmokeTestDockerfiles(EXECUTABLE_PKG_INFO)) { %>
  - name: pkg_test_<%= asTaskName('pkg_test', dockerfile) %>
<% } %>
```

Comment (addaleax) [reply .evergreen/evergreen.yml.in:1172]: Mh yeah, we could look into using `lodash.template` at some point, and we can also add helper functions for simplifying this.

It’s definitely easier not to outsource everything from the YAML here though; it has been very helpful to have a single source of truth for how our CI jobs are defined.

**Review by mcasimir — APPROVED** [review]
Awesome! If you have some time and give a thought on whether we could simplify the yaml generation that would be cool, but if not looks good to go!

File: .evergreen/evergreen.yml.in (hunk 4 above)
Comment (mcasimir) [reply .evergreen/evergreen.yml.in:1172]: Yeah, that's a good point, I mainly find it hard to parse, probably just having helpers to make the code "lighter" would be enough.

File: scripts/docker/rocky8-rpm.Dockerfile (hunk 3 above)
Comment (addaleax) [reply scripts/docker/rocky8-rpm.Dockerfile:8]: me neither :)

File: .evergreen/evergreen.yml.in (hunk 4 above)
Comment (addaleax) [reply .evergreen/evergreen.yml.in:1172]: It absolutely is hard to parse! I don’t want to act like it isn’t :) 

I’ll definitely think about it, for now my main goal here is getting everything to work tbh
